Rubberduck.Core/UnitTesting/TestMethod.cs

```diff
@@ -77,6 +78,17 @@ public TestResult Result
             set { _result = value; OnPropertyChanged(); }
         }
 
+        public string Category
+        {
+            get
+            {
+                var testMethodAnnotation = (TestMethodAnnotation) Declaration.Annotations
+                    .First(annotation => annotation.AnnotationType == AnnotationType.TestMethod);
```

Comment:
Shouldn't this be `FirstOrDefault` to avoid exception thrown when there's no such annotations?

Reply:
I agree with the assumption going on here (a test method has a `@TestMethod` annotation), but I feel like the tests need to confirm what happens when a method has e.g. two such annotations. IIRC we only consider the first one immediately on top of the method anyway, but I wouldn't know for sure without digging up the code for it. If a test method can only have a single `@TestMethod` annotation, then that should probably be a `.Single(annotation => annotation.AnnotationType == AnnotationType.TestMethod)` call, made `.SingleOrDefault` just to avoid having a property getter than can conceivably throw an `InvalidOperationException`. Or am I over-thinking this?

Reply:
On that note: Should we update the IllegalAnnotationInspection to check our assumptions then? I think we can not guarantee that there will only ever be one ` @TestMethod` annotation for any given Test Method. What about multiple categories?

Reply:
@Vogel612 good point about multiple categories. Maybe not in *this* PR, but this should eventually be supported:

    '@TestMethod("Category1", "Category2")

Not *illegal* per se, but should definitely fire a result or something that tells the user why the 2nd category is ignored. Then again, might be a bit out of scope for this specific PR.

Reply:
> If a test method can only have a single `@TestMethod` annotation, then that should probably be a `.Single(annotation => annotation.AnnotationType == AnnotationType.TestMethod)` call, made `.SingleOrDefault`

I have to disagree. Having multiple `@TestMethod` annotations does not produce a parse error, which means we will be throwing exceptions when nothing is wrong with the code from user's perspective. Maybe it would be better to take the first annotation that does have a non-whitespace category?

Reply:
OTOH, what does it mean to have multiple `@TestMethod`? The only thing I can imagine is that I screwed up something. In which case, I need to be reminded that I did truly screw up and need to take corrective action. Doing nothing and silently ignoring the redundant annotation only hides that fact that I blundered. Besides, raising an illegal annotation shouldn't mean throwing an exception; it should show up in the inspections, no?

Reply:
I take back what I said about the exception - of course `SingleOrDefault` does not throw anything. I don't know how duplicate annotations are handled at the moment, but I think we could go for ignoring all annotations except the first one + having a `DuplicateAnnotationInspection` inspection with `Error` severity. Having this inspection would warrant the use of `First` because the user will be notified that only the first annotation will be acknowledged (I am assuming that annotations are added to the collection in order of appearance in code).